crypto/evp/e_aes.c

```diff
@@ -950,6 +950,727 @@ static const EVP_CIPHER aes_##keylen##_##mode = { \
 const EVP_CIPHER *EVP_aes_##keylen##_##mode(void) \
 { return SPARC_AES_CAPABLE?&aes_t4_##keylen##_##mode:&aes_##keylen##_##mode; }
 
+#elif defined(OPENSSL_CPUID_OBJ) && defined(__s390__)
+/*
+ * IBM S390X support
+ */
+# include "s390x_arch.h"
+
+typedef struct {
+    union {
+        double align;
```

Comment:
Note that you don't actually have to have this align field, because structure contains a 64-bit field. I mean structure is always aligned at largest field. In other places where it's used, structures don't have 64-bit fields [not on 32-bit platforms]. This is a side note, i.e. action is not actually expected.

But on related note, could you add a note that following layout is what hardware expects. I mean that fields' order and sizes are not matter of our choice. Drop even line of reference to spec, e.g. "see KMA-GCM-AES Functions in Principles of Operation" so that it can be easily located with string search.

Reply:
the align member is not needed, right. Id like to keep it for it will still ensure proper alignment even in case the layout is realized using u8 only.

Ill add the ref to POP.